aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -703,6 +714,27 @@ void gemm_and_bias(
       &workspaceSize,
       sizeof(workspaceSize)));
 
+  uint32_t a_alignment = _getAlignment(reinterpret_cast<uintptr_t>(mat1_ptr));
+  uint32_t b_alignment = _getAlignment(reinterpret_cast<uintptr_t>(mat2_ptr));
+  uint32_t c_alignment = _getAlignment(reinterpret_cast<uintptr_t>(result_ptr));
+  uint32_t d_alignment = _getAlignment(reinterpret_cast<uintptr_t>(bias));
+  TORCH_CUDABLAS_CHECK(cublasLtMatmulPreferenceSetAttribute(
```

Comment:
can we add tests when A, B and C have different alignments to demonstrate that it works?

Reply:
That test was added in #92201, this PR keeps the test but removes the sketchy workaround that was used  https://github.com/pytorch/pytorch/pull/92201/files#diff-3f31c52b48cfddf8f4617d809f7695b2e4a1c78656f8c4b5143a4b45d01fcf23